content/reference-transaction-format.md

```diff
@@ -25,6 +26,63 @@ Any signature type can authorize any type of transaction, with the following exc
 * Only the master key can [permanently give up the ability to freeze](concept-freeze.html#no-freeze).
 * You can never remove the last method of signing transactions from an address.
 
+
+## Understanding Master and Regular Keys
+
+In the XRP Ledger, a digital signature proves that a transaction is authorized to do a specific set of actions. A digital signature is created based on a key pair associated with the transaction's sending account. There are two types of key pairs available: master keys and regular keys. Each key pair includes a private key and a public key.
+
+***TODO: I feel like some concept docs on private and public keys would be great -- like when to use them. Talk about the master_key, master_seed, master_seed_hex, public_key, and public_key_hex, for example, and why we provide them and what are the use cases. It would also be good to provide concept doc on how keys are used to generate the "signature" and the different types of signature-like values that we generate: Signature, SigningPubKey, and TxnSignature, for example. What are the use cases? I believe this info is in the doc, but discussed in the context of tutorials.***
```

Comment:
Reasonable. Here's my breakdown (for the record, "secret key" and "private key" are synonyms):

In `wallet_propose`:

| Field | Meaning | Use Case |
|---|---|---|
| `account_id` | The address of an account (or potential account), in base58 format. | This is not the public key, but a hash-of-a-hash of it. It also has a checksum so a typo almost certainly results in an invalid address rather than a valid but different address. This is the primary identifier of an account in the XRP Ledger. You tell people this to get paid, and use it in transactions to indicate who you are and who you're paying, trusting, etc. Multi-signing lists also use these to identify other signers. |
| `master_key` | The private key in [RFC-1751 format](https://tools.ietf.org/html/rfc1751). | An easier to remember, easier-to-write-down version of the private key. Can be used to sign transactions. |
| `master_seed` | The private key in base58 format. | A compact version of the private key. Typically you use the key in this format to sign transactions. I don't know if these are encoded with a checksum, but if they are, then that's an extra benefit—a typo results in an invalid key, not a valid but different key. |
| `master_seed_hex` | The private key in hexadecimal format. | A simple, widely-supported way to represent the secret key. You can also sign transactions with this format. |
| `public_key` | The public key in base58 format. | Not very useful, TBH. Like other base58 encodings, it's easy to copy-paste and doesn't have lookalikes capital i and lowercase L, or zero and uppercase letter o. |
| `public_key_hex` | The public key in hex format. | The public key. You need this to validate the signature on a transaction (or any signature). When you send a transaction, it actually attaches the public key so you can do so. (See `SigningPubKey`) |


| Field | Meaning | Use Case |
|---|---|---|
| `SigningPubKey` (top-level) | The public key of the sender in hex format. | To verify whether a transaction is valid, you check that all of the following are true: (a) this key hashes to an address that's authorized by the transaction's sender¹, (b) this key matches the signature on the transaction, (c) the signature matches the transaction instructions. Empty in the case of a multi-signed transaction. |
| `Signature` | The signature of the transaction instructions, in hex format. | Used with the `SigningPubKey` to verify that the sender did in fact approve the corresponding transaction instructions. |
| `Signers` (in a multi-signed tx) | An array of signature data for a multi-signed transaction. | Used to verify that a quorum of signers approved a transaction. |
| `Signers[*].AccountID` | The address of one signer, in base58 | Identifies which signer from the (predefined) multi-sigining list this portion of the multi-signature represents |
| `Signers[*].TxnSignature` | One signature, as hexadecimal. | Verifying a multi-signed transaction involves making sure each such signature is valid for its SigningPubKey and the transaction instructions. |
| `Signers[*].SigningPubKey` | The public key of one signer. | Verifying a multi-signed transaction involves making sure each such key is authorized to sign for the AccountID of the signer². |

¹ The default is that only "the address of an account" is authorized to send all transactions for that account. That address is the hash-of-a-hash-with-a-checksum of the public key from the master key pair that was generated during address creation. Regular keys add a different address (derived from a different key pair) that's authorized to send most transactions. And of course, you can also disable the master key or add a multi-signing list.

² Multi-signature AccountIDs are a little special. If one is an address that doesn't exist in the ledger, then the SigningPubKey must hash to the AccountID value. If the address _does_ exist in the ledger, then either (a) the SigningPubKey must hash to the AccountID _and_ the address must not have the master key disabled, or (b) the SigningPubKey must hash to a regular key that the address has set in the ledger.

Reply:
Thank you, above and beyond, @mDuo13! I added the wallet_propose field descrs to the wallet_propose doc. I created an Understanding Signatures concept doc with the signatures info you provided.